raiden-ts/src/transfers/epics.ts

```diff
@@ -80,101 +81,223 @@ function nextNonce(balanceProof?: SignedBalanceProof): UInt<8> {
+    nonce: nextNonce(channel.own.balanceProof),
+    transferred_amount: (channel.own.balanceProof
+      ? channel.own.balanceProof.transferredAmount
+      : Zero) as UInt<32>,
+    locked_amount: (channel.own.balanceProof ? channel.own.balanceProof.lockedAmount : Zero).add(
+      lock.amount,
+    ) as UInt<32>,
+    locksroot,
+    payment_identifier: action.payload.paymentId,
+    token,
+    recipient,
+    lock,
+    target: action.payload.target,
+    initiator: address,
+    metadata,
+  };
+  return from(signMessage(signer, message)).pipe(
+    mergeMap(function*(signed) {
+      // besides transferSigned, also yield transferSecret (for registering) if we know it
+      if (action.payload.secret)
+        yield transferSecret({ secret: action.payload.secret }, action.meta);
+      yield transferSigned({ message: signed, fee }, action.meta);
+      // messageSend LockedTransfer handled by transferSignedRetryMessageEpic
+    }),
+  );
+}
+
 /**
  * Create an observable to compose and sign a LockedTransfer message/transferSigned action
- * As it's an async observable which depends on state and may return an action which changes it,
- * the returned observable must be subscribed in a serialized context that ensures non-concurrent
- * write access to the channel's balance proof (e.g. concatMap)
+As it's an async observable which depends on state and may return an action which changes it,
+the returned observable must be subscribed in a serialized context that ensures non-concurrent
+write access to the channel's balance proof (e.g. concatMap)
  *
  * @param state$ - Observable of current state
  * @param action - transfer request action to be sent
- * @param network,address,signer - RaidenEpicDeps members
+ * @param deps - RaidenEpicDeps
  * @returns Observable of transferSigned|transferSecret|transferFailed actions
  */
 function makeAndSignTransfer(
   state$: Observable<RaidenState>,
   action: ActionType<typeof transfer>,
-  { network, address, signer, config$ }: RaidenEpicDeps,
+  deps: RaidenEpicDeps,
 ) {
-  return combineLatest([state$, config$]).pipe(
+  return combineLatest([state$, deps.config$]).pipe(
     first(),
-    mergeMap(([state, { revealTimeout }]) => {
-      if (action.meta.secrethash in state.sent) {
-        // don't throw to avoid emitting transferFailed, to just wait for already pending transfer
-        console.error('transfer already present', action.meta);
-        return EMPTY;
-      }
-
-      // assume paths are valid and recipient is first hop of first route
-      // compose metadata from it, and use first path fee
-      const metadata: Metadata = {
-          routes: action.payload.paths.map(({ path }) => ({ route: path })),
-        },
-        fee = action.payload.paths[0].fee,
-        recipient = action.payload.paths[0].path[0];
+    mergeMap(([state, { revealTimeout }]) =>
+      makeAndSignTransfer$(state, action, revealTimeout, deps),
```

Comment:
Maybe it would be better to pass the whole `config` to the inner function,a s `state` & `action`, and unpack there, instead of a specific `revealTimeout`